### DISCUSSION TITLE
NodaTime not mapping types?

### DISCUSSION THREAD

**Discussion by timbze** [linq2db/linq2db#4502] [General]
First, I am trying this in linq select: `i.Date.InZone(timeZone).Year`

The error is: `LinqToDB.Common.LinqToDBConvertException: Cannot convert value '2024-03-31T04:28:47Z: NodaTime.Instant' to type 'System.Int32'`

Linq query:
```c#
var sales = await _context.Document
    .Where(i => i.CustomerId == request.Id)
    .Where(i => i.Date >= startDate)
    .GroupBy(i => new{i.Date.InZone(timeZone).Year, i.Date.InZone(timeZone).Month})
    .OrderByDescending(i => i.Key.Year)
    .ThenByDescending(i => i.Key.Month)
    .Select(g => new CustomerMonthlySalesApiRs
    {
        Year = g.Key.Year,
        Month = g.Key.Month,
        Sales = g.Sum(i => i.TotalWithoutTax)
    })
    .ToListAsync();
```

The query output:
```sql
SELECT
	i."date",
	Sum(i.total_without_tax)
FROM
	document i
WHERE
	i.customer_id = :CustomerId AND
	i."date" >= :startDate
GROUP BY
	i."date"
ORDER BY
	i."date" DESC
```

According to [these Npgsql docs](https://www.npgsql.org/efcore/mapping/nodatime.html#operation-translation), it seems like it should work. What am I missing?

I do have this in my datasource builder in connection factory:
```c#
var dsBuilder = new NpgsqlDataSourceBuilder(connectionString);
dsBuilder.UseNodaTime();
```

**Comment by sdanyliv** [comment]
It is not supported yet. `Npgsql EF Core` is EF Core provider, but `linq2db` is not EF Core and has to know how to convert `NodaTime` methods.

**Reply by timbze** [reply]
Ok thank you
